agent/engine/docker_container_engine.go

```diff
@@ -136,7 +142,8 @@ func (dg *DockerGoClient) pullImage(image string) DockerContainerMetadata {
 	// Workaround for devicemapper bug. See:
 	// https://github.com/docker/docker/issues/9718
 	pullLock.Lock()
-	defer pullLock.Unlock()
+	ttime.Sleep(utils.AddJitter(0, dockerPullJitter))
```

Comment:
The workaround already relied on no third party pulling at the same time either... We should try and get some numbers on how good or bad jitter vs mutex is here.